src/components/encoder/Encoder.ts

```diff
@@ -0,0 +1,25 @@
+import type { Extra, PositionType } from '../../types';
+
+export interface Properties {
+  /** Whether a encoder supports ticks and/or degrees. */
+  ticksCountSupported: boolean;
+  angleDegreesSupported: boolean;
+}
+
+/** Represents a physical encoder. */
+export interface Encoder {
+  /** Set the current position of the encoder as the new zero position. */
+  resetPosition(extra?: Extra): Promise<void>;
+
+  /** Return the encoder's properties. */
+  getProperties(extra?: Extra): Promise<Properties>;
```

Comment:
if you agree with re-exporting the type

```suggestion
  getProperties(extra?: Extra): Promise<EncoderProperties>;
```